src/Development/Shake/Command.hs

```diff
@@ -145,7 +145,8 @@ commandExplicit funcName oopts results exe args = do
         shelled = runShell (unwords $ exe : args)
 
         ignore = map (?==) shakeLintIgnore
-        ham cwd xs = [makeRelative cwd x | x <- map toStandard xs
+        ham cwd xs = liftIO $ sequence [fromMaybe x <$> makeRelativeEx cwd x
```

Comment:
What does `ham` actually mean?
There is an underlying assumption here that track uses relative paths as keys. Is that correct? It's troubling to me that there may be many ways to refer to the same file (relative, absolute, via different symlinks or indirections), yet there doesn't seem to be any mechanism to enforce a canonical usage.